trunner/runners/STM32L4Runner.py

```diff
@@ -0,0 +1,77 @@
+#
+# Phoenix-RTOS test runner
+#
+# Stm32l4 runner
+#
+# Copyright 2021 Phoenix SYstems
+# Authors: Mateusz Niewiadomski
+#
+
+import serial
+import time
+import pexpect.fdpexpect
+
+import sys
+from os import path
+from subprocess import Popen, PIPE
+
+from .common import DeviceRunner
+from .common import _BOOT_DIR
+
+
+class STM32L4Runner(DeviceRunner):
+
+    # oneByOne_fdspawn class is a workaround for Phoenix-RTOS on stm32l4 targets not processing characters fast enough
+    # This class inherits and is passed to harness instead of pexpect.fdpexpect.fdspawn
+    # It redefines send() with addition of timeout after each letter sent. Timeout set to 30ms per character
+    class oneByOne_fdspawn(pexpect.fdpexpect.fdspawn):
+
+        def send(self, s):
+            r = 0
+            for c in s:
+                r += super().send(c)
+                time.sleep(0.03)
+            return r
+
+    # STM32L4Runner class variables:
+    # - BINARY_NAME: name of Phoenix-RTOS image binary that should be available in _boot/ directory
+    # - OPENOCD_CMD: string with shell command calling openocd with correct parameters to flash target
+    OPENOCD_CMD = ['openocd -f interface/stlink.cfg -f target/stm32l4x.cfg -c \
+"reset_config srst_only srst_nogate connect_assert_srst" -c "program ', ' 0x08000000 verify reset exit"']
+    BINARY_PATH = path.join(_BOOT_DIR, 'phoenix-armv7m4-stm32l4x6.bin')
+
+    def __init__(self, port1):
+        super().__init__(port1)
+
+    def flash(self):
+        # Flashing with openocd as a separate process
+        command = self.OPENOCD_CMD[0] + self.BINARY_PATH + self.OPENOCD_CMD[1]
+        openocd_process = Popen(
+            command,
+            shell=True,
+            stdout=PIPE,
```

Comment:
we don't want to catch openocd stdout? asking - I don't know what openocd prints